### PR TITLE
allow local kbfs mount use

### PR DESCRIPTION
**Pull request by zanderz**
I tried this on Linux. Won't work on Windows.

### CODE REVIEW

File: go/client/cmd_simplefs.go [hunk 1]
```diff
@@ -69,6 +69,12 @@ func makeSimpleFSPath(g *libkb.GlobalContext, path string) keybase1.Path {
 
 	path = filepath.ToSlash(filepath.Clean(path))
 
+	// Certain users seem to want to use SimpleFS on their
+	// mounted KBFS
+	if strings.HasPrefix(path, mountDir) {
+		return keybase1.NewPathWithKbfs(path[len(mountDir):])
+	}
```
Comment (oconnor663) [inline go/client/cmd_simplefs.go]: This seems duplicated from line 55 above. Is that intended?

Also it looks like this is relying on `mountDir` not having a trailing slash. Is that something we can rely on?

Comment (zanderz) [inline go/client/cmd_simplefs.go]: `keybase/[...]` fails the `IsAbs()` test on Windows, so we have to test it there too.

`mountDir` is a constant we define, so yes.

I added comments and made mountDir a constant, and addressed some lint errors.

**Comment by zanderz** [comment]
@oconnor663 ping!

File: go/client/cmd_simplefs.go [hunk 2]
```diff
@@ -44,14 +44,17 @@ func NewCmdSimpleFS(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Comm
...
 func makeSimpleFSPath(g *libkb.GlobalContext, path string) keybase1.Path {
-	mountDir := "/keybase"
 
 	path = filepath.ToSlash(path)
 	if strings.HasSuffix(path, "/") {
 		path = path[:len(path)-1]
 	}
 
+	// Test for the special mount dir prefix before the absolute test.
+	// Otherwise the current dir will be prepended, below.
```
Comment (oconnor663) [inline go/client/cmd_simplefs.go]: If `mountDir` is a prefix of `path`, though, doesn't that mean `path` has to be absolute? Shouldn't the current-dir-prepending on line 65 get skipped in that case?

File: go/client/cmd_simplefs.go [hunk 3]
```diff
@@ -44,14 +44,17 @@ func NewCmdSimpleFS(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Comm
...
 
+const mountDir = "/keybase"
+
 func makeSimpleFSPath(g *libkb.GlobalContext, path string) keybase1.Path {
-	mountDir := "/keybase"
 
 	path = filepath.ToSlash(path)
 	if strings.HasSuffix(path, "/") {
 		path = path[:len(path)-1]
 	}
```
Comment (oconnor663) [inline go/client/cmd_simplefs.go]: `filepath.Clean` can take care of this, and it gets called automatically by `Abs` and `EvalSymlinks`.

File: go/client/cmd_simplefs.go [hunk 4]
```diff
@@ -70,6 +73,13 @@ func makeSimpleFSPath(g *libkb.GlobalContext, path string) keybase1.Path {
```
Comment (oconnor663) [inline go/client/cmd_simplefs.go]: The `filepath.Abs` function can do the `IsAbs` + `Join` thing above.

File: go/client/cmd_simplefs.go [hunk 5]
```diff
@@ -70,6 +73,13 @@ func makeSimpleFSPath(g *libkb.GlobalContext, path string) keybase1.Path {
 
 	path = filepath.ToSlash(filepath.Clean(path))
 
+	// Certain users seem to want to use SimpleFS on their
+	// mounted KBFS. This is for those who want to do so
+	// from "/keybase/..."
+	if strings.HasPrefix(path, mountDir) {
+		return keybase1.NewPathWithKbfs(path[len(mountDir):])
```
Comment (oconnor663) [inline go/client/cmd_simplefs.go]: `strings.TrimPrefix` might be cleaner. And just to make sure I'm reading right, we want the with-KBFS paths to start with a slash, even though they're missing the mount dir?

File: go/client/cmd_simplefs.go (hunk 2 above)
Comment (zanderz) [inline go/client/cmd_simplefs.go]: Not on Windows

File: go/client/cmd_simplefs.go (hunk 3 above)
Comment (zanderz) [inline go/client/cmd_simplefs.go]: Same deal. Clean prepends the current dir on Windows.

File: go/client/cmd_simplefs.go (hunk 5 above)
Comment (zanderz) [inline go/client/cmd_simplefs.go]: There has been a lot of discussion about identifying when the local dir is actually a KBFS mount, and there is not a really solid way to do it on all the platforms. For now, a path starting with "/keybase" is KBFS.

Comment (zanderz) [inline go/client/cmd_simplefs.go]: Also, we need the result of the test to distinguish what kind of path to make - otherwise I agree.

**Pull request merged**